server/src-lib/Hasura/GraphQL/Transport/HTTP.hs

```diff
@@ -42,31 +42,27 @@ runGQ reqId userInfo reqHdrs req = do
         E.Tree resolvedPlan unresolvedPlansNE -> do
           let unresolvedPlans = toList unresolvedPlansNE -- it's safe to convert here
           HttpResponse initJson _ <- runLeafPlan resolvedPlan
-          let remoteRels =
-                map
-                  (\(E.QExecPlanUnresolved remoteRelField _) -> remoteRelField)
-                  unresolvedPlans
           let (initValue, remoteBatchInputs) =
-                E.extractRemoteRelArguments
-                  initJson
-                  remoteRels
+                E.extractRemoteRelArguments initJson $
+                  map E.remoteRelField unresolvedPlans
+
+          -- TODO This zip may discard some unresolvedPlans when permissions
```

Comment:
what permissions do you mean?

Reply:
For instance in `basic_relationship_with_permissions1.yaml`, where we filter a field to null or an empty list; the two lists have unequal length there.